### PR TITLE
Add stats update client call.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/29/
Test PASSed.

File: csclient/params/params.go [hunk 1]
```diff
@@ -44,6 +44,21 @@ type ArchiveUploadResponse struct {
 	PromulgatedId *charm.Reference `json:",omitempty"`
 }
 
+const (
+	Download = "download"
```
Comment (urosj) [inline csclient/params/params.go]: Add description.

File: csclient/csclient_test.go [hunk 2]
```diff
@@ -304,6 +304,17 @@ func (s *suite) TestGetArchiveWithStatsDisabled(c *gc.C) {
 	s.checkCharmDownloads(c, key, 0)
 }
 
+//func (s *suite) TestStatsUpdate(c *gc.C) {
```
Comment (urosj) [inline csclient/csclient_test.go]: This will be uncommented later, right?

File: csclient/params/params.go [hunk 3]
```diff
@@ -44,6 +44,21 @@ type ArchiveUploadResponse struct {
...
+	Traffic  = "traffic"
```
Comment (mhilton) [inline csclient/params/params.go]: What's a traffic mean? perhaps these need a comment.

**Comment by mhilton** [comment]
:+1:

File: csclient/params/params.go [hunk 4]
```diff
@@ -44,6 +44,21 @@ type ArchiveUploadResponse struct {
...
+	Deploy   = "deploy"
+)
+
+// StatsUpdateRequest holds the paramaeters for a put to /stats/update.
+// See https://github.com/juju/charmstore/blob/v4/docs/API.md#stats-update
```
Comment (urosj) [inline csclient/params/params.go]: There's no such link in the docs.

**Comment by urosj** [comment]
Please add comments on consts, also add the description in the API.md. Then LGTM.

File: csclient/csclient_test.go (hunk 2 above)
Comment (fabricematrat) [inline csclient/csclient_test.go]: Yes once the charmstore equivalent will be ready.

File: csclient/params/params.go (hunk 4 above)
Comment (fabricematrat) [inline csclient/params/params.go]: Not yet as the charmstore equivalent is not delivered.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/30/
Test PASSed.

**Comment by fabricematrat** [comment]
:shipit:  thanks @mhilton && @urosj

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Pull request merged**